src/Compilers/Core/Portable/CodeGen/SequencePointList.cs

```diff
@@ -124,6 +124,35 @@ public ImmutableArray<Cci.SequencePoint> GetSequencePoints(DebugDocumentProvider
                 current = current._next;
             }
 
+            // Find the document for the first non-hidden sequence point (issue #4370)
+            current = this;
+            while (current != null)
```

Comment:
The reason I kept it inline is because I need to set multiple variables. I'm ok to extract a method with `out` parameters if that's what you recommend.